tests/unit/varint_tests.cpp

```diff
@@ -252,6 +252,194 @@ EOSIO_TEST_BEGIN(signed_int_type_test)
+   // friend bool operator==(const signed_int&, const uint32_t&)
+   static_assert( signed_int{42} == int32_t{42} );
+   static_assert( (signed_int{42} == int32_t{43}) == false );
+
+   // ----------------------------------------------------------
+   // friend bool operator==(const uint32_t&, const signed_int&)
+   static_assert( int32_t{42} == signed_int{42} );
+   static_assert( (int32_t{43} == signed_int{42}) == false );
+
+   // ------------------------------------------------------------
+   // friend bool operator==(const signed_int&, const signed_int&)
+   static_assert( signed_int{42} == signed_int{42} );
+   static_assert( (signed_int{42} == signed_int{43}) == false );
+
+   // ----------------------------------------------------------
+   // friend bool operator!=(const signed_int&, const uint32_t&)
+   static_assert( (signed_int{42} != int32_t{42}) == false );
+   static_assert( signed_int{42} != int32_t{43} );
+
+   // ----------------------------------------------------------
+   // friend bool operator!=(const uint32_t&, const signed_int&)
+   static_assert( (int32_t{42} != signed_int{42}) == false );
+   static_assert( int32_t{43} != signed_int{42} );
+
+   // ------------------------------------------------------------
+   // friend bool operator!=(const signed_int&, const signed_int&)
+   static_assert( (signed_int{42} != signed_int{42}) == false );
+   static_assert( signed_int{42} != signed_int{43} );
+
+   // ----------------------------------------------------------
+   // friend bool operator< (const signed_int&, const uint32_t&)
+   static_assert( (signed_int{42} < int32_t{42}) == false );
+   static_assert( signed_int{42} < int32_t{43} );
+
+   // ----------------------------------------------------------
+   // friend bool operator< (const uint32_t&, const signed_int&)
+   static_assert( (int32_t{42} < signed_int{42}) == false );
+   static_assert( int32_t{42} < signed_int{43} );
+
+   // ------------------------------------------------------------
+   // friend bool operator< (const signed_int&, const signed_int&)
+   static_assert( (signed_int{42} < signed_int{42}) == false );
+   static_assert( signed_int{42} < signed_int{43} );
+
+   // ----------------------------------------------------------
+   // friend bool operator>=(const signed_int&, const uint32_t&)
+   static_assert( signed_int{42} >= int32_t{42} );
+   static_assert( (signed_int{42} >= int32_t{43}) == false );
+
+   // ----------------------------------------------------------
+   // friend bool operator>=(const uint32_t&, const signed_int&)
+   static_assert( int32_t{42} >= signed_int{42} );
+   static_assert( (int32_t{42} >= signed_int{43}) == false );
+
+   // ------------------------------------------------------------
+   // friend bool operator>=(const signed_int&, const signed_int&)
+   static_assert( signed_int{42} >= signed_int{42} );
+   static_assert( (signed_int{42} >= signed_int{43}) == false );
```

Comment:
Nit, I'd add more 2,3 & 4 bytes int/uint tests with casts or integer literals i.e. `xU`, `xL` etc.. and `std::size_t`, just to make sure everything passes at compile time (which of course should).

Reply:
Thanks. More tests were added.